helix-dap/src/client.rs

```diff
@@ -0,0 +1,572 @@
+use crate::{
+    transport::{Event, Payload, Request, Response, Transport},
+    Result,
+};
+use log::{error, info};
+use serde::{Deserialize, Serialize};
+use serde_json::{from_value, to_value, Value};
+use std::sync::{
+    atomic::{AtomicU64, Ordering},
+    Arc,
+};
+use std::{collections::HashMap, process::Stdio};
+use tokio::{
+    io::{AsyncBufRead, AsyncWrite, BufReader, BufWriter},
+    net::TcpStream,
+    process::{Child, Command},
+    sync::{
+        mpsc::{channel, Receiver, Sender, UnboundedReceiver, UnboundedSender},
+        Mutex,
+    },
+};
+
+#[derive(Debug, PartialEq, Clone, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct DebuggerCapabilities {
+    pub supports_configuration_done_request: Option<bool>,
+    pub supports_function_breakpoints: Option<bool>,
+    pub supports_conditional_breakpoints: Option<bool>,
+    pub supports_exception_info_request: Option<bool>,
+    pub support_terminate_debuggee: Option<bool>,
+    pub supports_delayed_stack_trace_loading: Option<bool>,
+    // TODO: complete this
+}
+
+#[derive(Debug, PartialEq, Clone, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+struct InitializeArguments {
+    #[serde(rename = "clientID")]
+    client_id: Option<String>,
+    client_name: Option<String>,
+    #[serde(rename = "adapterID")]
```

Comment:
- [ ] Capitalize everything like so for exact name compatibility for all implementations

Reply:
Everything else looks OK. `threadId` is capitalized like this in docs.